src/pages/shipping.js

```diff
@@ -314,87 +314,119 @@ const handleRemoveBagButton = (currDeleteButton, currTubes, currBoxId) => {
+                return false;
+            }
     
-    let largestBoxId;
-    if (shouldUpdateBoxModal) largestBoxId = boxList[largestBoxIndex][conceptIds.shippingBoxId];
-    else largestBoxId = largestBoxIndex !== -1 ? boxList[largestBoxIndex][conceptIds.shippingBoxId] : 'Box0';
-
-    if (largestBoxIndexAtLocation == -1 || shouldUpdateBoxModal) {
-        const boxToAdd = await createNewBox(boxList, siteLocationConversion, siteCode, largestBoxId, shouldUpdateBoxModal);
-        if (!boxToAdd) {
-            showNotifications({ title: 'ERROR ADDING BOX - PLEASE REFRESH YOUR BROWSER', body: 'Error: This box already exists. A member of your team may have recently created this box. Please refresh your browser and try again.' });
+            document.getElementById('shippingModalChooseBox').setAttribute('data-new-box', boxToAdd[conceptIds.shippingBoxId]);
+            updateShippingStateCreateBox(boxToAdd);
+            return true;
+        } else {
             return false;
         }
-
-        updateShippingStateCreateBox(boxToAdd);
-        return true;
-    } else {
+    } catch (e) {
+        console.error('Error adding box', e);
+        showNotifications({
+            title: 'ERROR ADDING BOX',
+            body: 'An unexpected error occurred. Please try again later.'
+        });
         return false;
     }
 }
 
-const createNewBox = async (boxList, pageLocationConversion, siteCode, largestBoxId, updateBoxModal) => {
-    const newBoxNum = parseInt(largestBoxId.substring(3)) + 1;
-    const newBoxId = 'Box' + newBoxNum.toString();
+// Create the new box and add it to firestore. If the box already exists, wait one second, increment the boxId, and try again up to 3 times.
+const createNewBox = async (boxList, pageLocationConversion, siteCode, largestBoxNum, docId) => {
+    let attempts = 0;
+    let maxAttempts = 3;
+
     const boxToAdd = {
-        'bags': {},
-        [conceptIds.shippingBoxId]: newBoxId,
         [conceptIds.shippingLocation]: pageLocationConversion,
         [conceptIds.siteCode]: siteCode,
         [conceptIds.submitShipmentFlag]: conceptIds.no,
-        [conceptIds.siteShipmentReceived]: conceptIds.no
+        [conceptIds.siteShipmentReceived]: conceptIds.no,
+        ['siteDetailsDocRef']: docId,
     };
 
-    try {
-        const addBoxResponse = await addBox(boxToAdd);
-        if (addBoxResponse.message !== 'Success!') {
+    while (attempts < maxAttempts) {
+        largestBoxNum++;
+        boxToAdd[conceptIds.shippingBoxId] = 'Box' + largestBoxNum.toString();
+    
+        try {
+            const addBoxResponse = await addBoxAndUpdateSiteDetails(boxToAdd);
+            if (addBoxResponse.code === 200) {
+                boxList.push(boxToAdd);
+                return boxToAdd;
+            } else if (addBoxResponse.code === 409) {
+                attempts ++;
+                await delayRetryAttempt(1000);
```

Comment:
I like the one second delay for checking whether the next box# increment exists or not

Reply:
Thanks!